fvm/evm/handler/blockstore.go

```diff
@@ -76,11 +76,19 @@ func (bs *BlockStore) BlockProposal() (*types.BlockProposal, error) {
 	// expect timestamps in unix seconds so we convert here
 	timestamp := uint64(cadenceBlock.Timestamp / int64(time.Second))
 
+	// read a random value for block proposal
+	rand := gethCommon.Hash{}
+	err = bs.backend.ReadRandom(rand[:])
```

Comment:
I believe this would be problematic, since `BlockProposal()` is called multiple times in same execution:
- https://github.com/onflow/flow-go/blob/master/fvm/evm/handler/handler.go#L506
- https://github.com/onflow/flow-go/blob/master/fvm/evm/handler/handler.go#L387

And each call would produce different value before the block proposal is committed.

Reply:
damn, thats a good find, either we have to add a cache for the block proposal or as you suggested we use the cadenceBlock.Hash

Reply:
updated to use the hash as you suggested at [811a035](https://github.com/onflow/flow-go/pull/6378/commits/811a03575eebaa13c6825eab59dcb8514cfcf358)